### ISSUE TITLE
[SP9] Kenel panic prevents shutdown

### ISSUE THREAD

**Issue by plover43** [linux-surface/linux-surface#1113]
Not on every shutdown, maybe 1/10 times, shutdown hangs on the sp9. It seems as though it's caused by kernel panic. 

### Environment

- Hardware model: sp9 i5
- Kernel version: Linux hw 6.2.8-1.surface.fc37.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Mar 23 14:53:56 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
- Distribution: fedora 37

![kernel_panic](https://user-images.githubusercontent.com/129125889/229582420-f65cd4fc-746c-4a95-bb5a-e1e93c5f4a9e.jpg)

**Comment by peter-marshall5** [comment]
I can confirm that I'm also getting this. The reason for the panic is `Attempted to kill init`, indicating that it might be a problem with systemd. Strangely, the panic message also usually looks glitched / corrupted, with a few lines seeming to be partially replaced with text from other lines.

**Comment by theneosloth** [comment]
Still seeing this issue on 6.2.11-1.surface.fc38.x86_64. Have not been able to isolate a pattern.
The call trace seems more or less identical to OP

**Comment by qzed** [comment]
The patchset was removed in 6.2.11 on Fedora and 6.2.12 on Debian and Arch. So if the issue still happens it's very likely unrelated.

**Comment by CheeseFM** [comment]
Yeap! Just had that too on my SL5 while connected via USB-C, something with 13th gen again.

**Comment by qzed** [comment]
Interesting. I still have no clue though what could cause this.

**Comment by theneosloth** [comment]
Yeah I was able to replicate that as well on SP9 F38. Used the surface charging cable

**Comment by qzed** [comment]
So based on the logs it looks like the panic occurs here: https://elixir.bootlin.com/linux/v6.2.15/source/kernel/exit.c#L837. The `WARN_ON(irqs_disabled())` above also gets triggered, leading to the two stack traces.

**Comment by knard** [comment]
I don't know if it's the same issue but I got kernel panic during shutdown after installing TLP package on archlinux. Since uninstalled I don't have any kernel panic.

**Comment by derkrasseleo** [comment]
Can't confirm removing tlp solved hang on shutdown with SP9 and linux-surface 6.6.6

**Comment by dev-nis** [comment]
I had the same issue on my Surface Pro 9 running KDE Neon (based on Ubuntu 22.04). Removing the TLP package fixed it for me.
